src/bech32.cpp

```diff
@@ -147,11 +148,14 @@ namespace bech32
 std::string Encode(const std::string& hrp, const data& values) {
     data checksum = CreateChecksum(hrp, values);
     data combined = Cat(values, checksum);
-    std::string ret = hrp + '1';
-    ret.reserve(ret.size() + combined.size());
+    char ret[91];
+    memcpy(ret, hrp.c_str(), hrp.size());
+    char* ptr = ret + hrp.size();
+    *(ptr++) = '1';
     for (auto c : combined) {
-        ret += CHARSET[c];
+        *(ptr++) = CHARSET[c];
     }
+    *ptr = 0;
     return ret;
```

Comment:
Even faster: `return std::string(ret, ptr);` (no need for the string constructor to find the string length again, you already know).

Reply:
Damn.
```
Bech32Encode, 5, 800000, 3.79493, 9.32891e-07, 9.75863e-07, 9.47069e-07
```

Reply:
Actually, the vanity generator did not see an increase in speed so this may have been a fluke. It didn't make it worse though so keeping.

Reply:
Too bad it didn't help, but I prefer the avoidance of zero-terminated strings in any case.